src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AwaitExpressionSpiller.cs

```diff
@@ -854,6 +854,13 @@ public override BoundNode VisitConversion(BoundConversion node)
                     type: node.Type));
         }
 
+        public override BoundNode VisitMethodGroup(BoundMethodGroup node)
+        {
+            BoundSpillSequenceBuilder builder = null;
```

Comment:
>BoundSpillSequenceBuilder [](start = 12, length = 25)

LGTM

This fix is correct, but a method group should not get this far. This has exposed a flaw in lowering, which should turn the method group conversion into a delegate creation. I'll open a (design) issue.